cli/azd/pkg/infra/provisioning/bicep_provider.go

```diff
@@ -0,0 +1,452 @@
+				return
+			}
+
+			modulePath := p.modulePath()
+			asyncContext.SetProgress(&PreviewProgress{Message: "Compiling Bicep template", Timestamp: time.Now()})
+			template, err := p.createPreview(ctx, modulePath)
+			if err != nil {
+				asyncContext.SetError(fmt.Errorf("creating template: %w", err))
+				return
+			}
+
+			// Merge parameter values from template
+			for key, param := range template.Parameters {
+				if bicepParam, has := bicepTemplate.Parameters[key]; has {
+					param.Value = bicepParam.Value
+					template.Parameters[key] = param
+				}
+			}
+
+			result := PreviewResult{
+				Preview: *template,
+			}
+
+			asyncContext.SetResult(&result)
+		})
+}
+
+func (p *BicepProvider) UpdatePlan(ctx context.Context, preview Preview) error {
+	bicepFile := BicepTemplate{
+		Schema:         "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
+		ContentVersion: "1.0.0.0",
+	}
+
+	parameters := make(map[string]BicepInputParameter)
+
+	for key, param := range preview.Parameters {
+		parameters[key] = BicepInputParameter{
+			Type:         param.Type,
+			DefaultValue: param.DefaultValue,
+			Value:        param.Value,
+		}
+	}
+
+	bicepFile.Parameters = parameters
+
+	bytes, err := json.MarshalIndent(bicepFile, "", "  ")
+	if err != nil {
+		return fmt.Errorf("marshaling parameters: %w", err)
+	}
+
+	parametersFilePath := p.parametersFilePath()
+	err = ioutil.WriteFile(parametersFilePath, bytes, 0644)
+	if err != nil {
+		return fmt.Errorf("writing parameters file: %w", err)
+	}
+
+	return nil
+}
+
+// Provisioning the infrastructure within the specified template
+func (p *BicepProvider) Deploy(ctx context.Context, preview *Preview, scope Scope) *async.InteractiveTaskWithProgress[*DeployResult, *DeployProgress] {
+	return async.RunInteractiveTaskWithProgress(
+		func(asyncContext *async.InteractiveTaskContextWithProgress[*DeployResult, *DeployProgress]) {
+			isDeploymentComplete := false
+
+			err := asyncContext.Interact(func() error {
+				deploymentSlug := azure.SubscriptionDeploymentRID(p.env.GetSubscriptionId(), p.env.GetEnvName())
+				deploymentUrl := fmt.Sprintf("https://portal.azure.com/#blade/HubsExtension/DeploymentDetailsBlade/overview/id/%s\n\n", url.PathEscape(deploymentSlug))
+				err := p.console.Message(ctx, fmt.Sprintf("Provisioning Azure resources can take some time.\n\nYou can view detailed progress in the Azure Portal:\n%s", deploymentUrl))
+
+				if err != nil {
+					return err
+				}
+
+				return nil
+			})
+
+			if err != nil {
+				asyncContext.SetError(err)
+				return
+			}
+
+			// Start the deployment
+			go func() {
+				modulePath := p.modulePath()
+				parametersFilePath := p.parametersFilePath()
+				deployResult, err := p.deployModule(ctx, scope, modulePath, parametersFilePath)
+				var outputs map[string]PreviewOutputParameter
+
+				if err != nil {
+					asyncContext.SetError(err)
+					isDeploymentComplete = true
```

Comment:
You can `defer()` this at the beginning so you do not miss one of the return paths. Also, technically, this is also a race, in the sense that a write to `isDeploymentComplete` can precede result setting.

Reply:
Fixed.